### PR TITLE
Adds `category` field as an annotation as `tekton.dev/categories`

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- Initially all tags were mapped to categories in Hub, for e.g. [config.yaml](https://github.com/tektoncd/hub/blob/master/config.yaml),
    so whenever a new tag was added in a task it was mapped to a category called `others`.
    Hence before every release we had to manually map these new tags to some category,
    hence after the discussion in Catalog and Hub WG, a proposal was created for adding
    a category as an annotation.

  - PR to update the [TEP-0003-Tekton Catalog Organization](https://github.com/tektoncd/community/pull/352)
  - [Predefined Category List](https://docs.google.com/document/d/1lR2YfyOK-eQqgMncbw3tbymfp6tEDfWKLr53rIZsfo8/edit)

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by chmouel** [comment]
/hold 

let's wait for the TEP to pass on before going over this,

**Comment by afrittoli** [comment]
I think it would be good to have the list of categories stored in the repo and checked by CI.
I see several errors reported by Catlin: shouldn't the CI job fail when ERRORs are reported?

**Comment by PuneetPunamiya** [comment]
> I think it would be good to have the list of categories stored in the repo and checked by CI.
> I see several errors reported by Catlin: shouldn't the CI job fail when ERRORs are reported?

Yes I already have a pr in plumbing repo where Catlin will have a check and the CI would fail and will list all the predefined categories https://github.com/tektoncd/plumbing/pull/748

**Review by vdemeester — APPROVED** [review]

**Comment by bobcatfish** [comment]
those are the longest github comments ive ever seen @_@

**Comment by vinamra28** [comment]
> those are the longest github comments ive ever seen @_@

All credit goes to catlin :stuck_out_tongue:

File: task/gcloud/0.1/gcloud.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: CLI
```
Comment (vinamra28) [inline task/gcloud/0.1/gcloud.yaml:9]: how about adding one more category as Cloud along with CLI?

File: task/git-cli/0.1/git-cli.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/version: "0.1"
   annotations:
+    tekton.dev/categories: Git
```
Comment (vinamra28) [inline task/git-cli/0.1/git-cli.yaml:9]: here `Git, CLI` ?

File: task/github-create-deployment/0.1/github-create-deployment.yaml [hunk 3]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/version: "0.1"
   annotations:
+    tekton.dev/categories: Git
```
Comment (vinamra28) [inline task/github-create-deployment/0.1/github-create-deployment.yaml:9]: `Git, Deployment` ?

File: task/gke-deploy/0.1/gke-deploy.yaml [hunk 4]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Deployment
```
Comment (vinamra28) [inline task/gke-deploy/0.1/gke-deploy.yaml:9]: `Kubernetes, Deployment`?

File: task/ibmcloud/0.1/ibmcloud.yaml [hunk 5]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Cloud
```
Comment (vinamra28) [inline task/ibmcloud/0.1/ibmcloud.yaml:9]: `Cloud, CLI`?

File: task/kubeconfig-creator/0.1/kubeconfig-creator.yaml [hunk 6]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Deploymment, Kubernetes
```
Comment (vinamra28) [inline task/kubeconfig-creator/0.1/kubeconfig-creator.yaml:9]: `Kubernetes`?

File: task/kubectl-deploy-pod/0.1/kubectl-deploy-pod.yaml [hunk 7]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Kubernetes
```
Comment (vinamra28) [inline task/kubectl-deploy-pod/0.1/kubectl-deploy-pod.yaml:9]: `Kubernetes, Deployment`?

File: task/orka-deploy/0.1/orka-deploy.yaml [hunk 8]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/version: "0.1"
   annotations:
+    tekton.dev/categories: Deployment
```
Comment (vinamra28) [inline task/orka-deploy/0.1/orka-deploy.yaml:9]: `Cloud, Kubernetes`?

File: task/scorecard/0.1/scorecard.yaml [hunk 9]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/version: "0.1"
   annotations:
+    tekton.dev/categories: Git
```
Comment (vinamra28) [inline task/scorecard/0.1/scorecard.yaml:9]: `Security`?

File: task/skopeo-copy/0.1/skopeo-copy.yaml [hunk 10]
```diff
@@ -6,6 +6,7 @@ metadata:
     app.kubernetes.io/version: "0.1"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: CLI
```
Comment (vinamra28) [inline task/skopeo-copy/0.1/skopeo-copy.yaml:9]: `CLI, Image-Build`?

File: task/terraform-cli/0.2/terraform-cli.yaml [hunk 11]
```diff
@@ -7,6 +7,7 @@ metadata:
     app.kubernetes.io/version: "0.2"
   annotations:
     tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: CLI
```
Comment (vinamra28) [inline task/terraform-cli/0.2/terraform-cli.yaml:10]: `CLI, Automation`?

**Review by vinamra28 — COMMENTED** [review]
@PuneetPunamiya thanks for doing this :+1:. Just added a couple of thoughts.
/cc @tektoncd/catalog-maintainers

**Comment by PuneetPunamiya** [comment]
cc @afrittoli  @bobcatfish  @jerop

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Pull request merged**